plugins/uf2/fu-uf2-device.c

```diff
@@ -157,6 +157,9 @@ fu_uf2_device_write_firmware(FuDevice *device,
 		return FALSE;
 	}
 
+	/* the device automatically reboots */
+	fu_device_add_flag(device, FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG);
```

Comment:
I think that's true, but only if `if (fu_device_has_flag(device, FWUPD_DEVICE_FLAG_WILL_DISAPPEAR))` is set.

Reply:
I was pondering that. Having spot checked the devices in uf2.quirk I believe they are all bootloaders, and will exhibit this behaviour. I'm not sure what the grand design is with will-disappear vs. is-booloader.

Maybe all the devices in uf2.quirk should be tagged will-disappear then?

Reply:
Ahh, I've re-read the original commit. What will-disappear means is "there might not be a runtime" i.e. if you flash a firmware to a UF2 device without a runtime USB interface from a fwupd point of view it just disappears.

Of course, the generic bootloader device is the wrong place to define if a runtime exists. :)

What I think we need is a way for the runtime (before it goes into bootloader) to pass to the bootloader uf2 mode to say "wait to come back into runtime".

I think we should probably remove `will-disappear` from the quirk file as it does nothing now and it's the wrong place. Maybe we could set the `use-runtime-version` in the runtime plugin and then propagate that to the uf2 device - and then if that's set we set `wait-for-replug` at the end of write-firmware.

Reply:
I think it would be good for the bootloader device (quirks) to know if the bootloader will automatically reset. Because thats likely (?) a property of the bootloader, and not of the runtime.

But I agree, `will-disappear` is the wrong flag (with your explanation).

I can introduce a new flag `will-automatically-reset`? It seems like a lot of devices already need this behaviour anyway?

Reply:
I'll draft something with `will-automatically-reset` as a private flag. Don't think it needs to be public

Reply:
force-pushed; this works fine for me

Reply:
Alternate proposal in https://github.com/fwupd/fwupd/pull/7936